### PR TITLE
feat: Add readOnly mode to Editor.ts

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: demos/gallery/src-umd/config.js [hunk 1]
```diff
@@ -724,6 +724,16 @@ var config = {
...
+          {
+            "path": "./samples/codemirror/ReadOnly.js",
+            "name": "ReadOnly.js",
+            "type": "file",
+            "imports": {
+              "@hpcc-js/codemirror": [
+                "JSEditor"
+              ]
+            }
+          },
```
Comment (GordonSmith) [inline demos/gallery/src-umd/config.js:736]: This file is missing?

Comment (jbrundage) [reply demos/gallery/src-umd/config.js:736]: I've added it.

File: packages/codemirror/src/Editor.ts [hunk 2]
```diff
@@ -9,12 +9,13 @@ export interface IPosition {
 }
 
 export class Editor extends HTMLWidget {
-    private _codemirror: CodeMirror.EditorFromTextArea;
+    protected _codemirror: CodeMirror.EditorFromTextArea;
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts:12]: This change (which is not needed) is causing the build failure.

Comment (jbrundage) [reply packages/codemirror/src/Editor.ts]: I noticed that too, had it changed to "protected" to make debugging easier. I've rolled it back and amended.

File: packages/codemirror/src/Editor.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export class Editor extends HTMLWidget {
 
     options(): any {
         return {
+            readOnly: this.readOnly(),
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts:18]: This should not be needed?

Comment (jbrundage) [reply packages/codemirror/src/Editor.ts:18]: That allows the option to be set at the point of instantiation.

Comment (GordonSmith) [reply packages/codemirror/src/Editor.ts:18]: Your code in the update method also gets called on instantiation?

Comment (jbrundage) [reply packages/codemirror/src/Editor.ts:18]: The codemirror instantiation takes place in the enter method: `CodeMirror.fromTextArea`

Then the update method sets the option on the already created instance of codemirror. That seems to work well enough with this readOnly option. It just seems like a good idea to allow codemirror to set the option before *their* instantiation.

**Pull request merged**